### ISSUE TITLE
DTA main site mobile / small screen menu issues

### ISSUE THREAD

**Issue by Decrepidos** [govau/dta-website-rebuild#697]
**DTA main site mobile / small screen menu**
Page tested: Simple, clear and fast public services
https://www.dta.gov.au/our-projects/digital-service-platforms-strategy

**Issues and recommendations**

Firstly I strongly advise against using ARIA menus for website navigation. These should only be used for Windows desktop type applications.

**Issues:**

**Role menu is missing child attribute role=menuitem.
Incorrect keyboard mapping.**

- Expected keyboard mapping for an ARIA menu
- Right Arrow: Moves focus to the next item in the menubar. If focus is on the last item, moves focus to the first item.
- Left Arrow: Moves focus to the previous item in the menubar.
- If focus is on the first item, moves focus to the last item.
- Down Arrow: Opens submenu and moves focus to first item in the submenu.
- Up Arrow: Opens submenu and moves focus to last item in the submenu.
- Home: Moves focus to first item in the menubar.
- End: Moves focus to last item in the menubar.
- Esc must close the expanded menu from anywhere on the page.

**Solution 1**: (recommended) Remove all existing ARIA menu attributes and roles: role=menu, aria-haspopup
Implement aria-expanded and aria-controls on the menu button. Aria controls to point to the sub menu container.

**Solution 2**:
Include role=menuitem on the focusable element in the expanded menu. <a href ...
Fix keyboard mapping.

Lastly the method used to label the navigation is not the best approach.

< nav role="navigation" aria-labelledby="main-navigation-label" >
< h2 class="visually-hidden" id="main-navigation-label">Main navigation

This is verbose and repetitive.
Screen readers will announce _'Main navigation navigation region' 'Heading level 2 Main navigation'._ Wow! can that get irritating fast.

**Solution:** Remove aria-labelledby and the visually-hidden H2. Implement a simple aria-label="Main" on <nav role="navigation"
And you don't need role=navigation on the < nav >

Cheers
Laurence

Bug report
[Yes ] I’ve read and understood the Contributing guidelines and have done my best effort to follow them.
[Yes ] I’ve read and agree to the Code of Conduct.
[ Yes] I’ve searched for any related issues and avoided creating a duplicate issue.\

**Comment by imperator99** [comment]
Hi @Decrepidos thanks for your issue.

I'm looking into this internally to come up with solution and will update the issue as I work on it.

**Comment by Decrepidos** [comment]
Hi @imperator99 Happy to answer any questions you may have.

**Comment by imperator99** [comment]
@Decrepidos I am working on this now (finally!). The HTML changes are straightforward, however is the keyboard navigation necessary? It is a simple menu with just the 7 items including the menu button itself, with no sub menus. It can be tabbed through with Enter used to select items.

**Comment by Decrepidos** [comment]
Hi @imperator99 My suggestions are best practice methods. In this instance if the menu roles are removed and it is setup as a native HTML5 construct  IMO tabbing only is sufficient. However, if the ARIA roles are retained the keyboard functionality should meet the expected keyboard navigation for an ARIA Menu. Thanks for keeping me in the loop. Cheers Laurence

**Comment by imperator99** [comment]
Excellent @Decrepidos then I think my solution will work nicely. Hopefully I can get it online shortly.

**Comment by imperator99** [comment]
Resolved by https://github.com/govau/dta-gov-au/pull/442 and https://github.com/govau/dta-website-rebuild/pull/737
